### ISSUE TITLE
Nginx errors on update

### ISSUE THREAD

**Issue by openoms** [raspiblitz/raspiblitz#3223]
```
₿ sudo apt update && sudo apt upgrade -y --fix-missing
Hit:1 http://archive.raspberrypi.org/debian bullseye InRelease
Hit:2 http://security.debian.org/debian-security bullseye-security InRelease                                                                             
Hit:3 http://deb.debian.org/debian bullseye InRelease                                                                                                    
Hit:4 http://deb.debian.org/debian bullseye-updates InRelease                                                                      
Get:5 https://pkgs.tailscale.com/stable/debian bullseye InRelease                                            
Get:6 tor+http://apow7mjfryruh65chtdydfmqfpj5btws7nbocgtaovhvezgccyjazpqd.onion/torproject.org bullseye InRelease [3,526 B]
Err:6 tor+http://apow7mjfryruh65chtdydfmqfpj5btws7nbocgtaovhvezgccyjazpqd.onion/torproject.org bullseye InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 74A941BA219EC810
Fetched 9,071 B in 3s (3,318 B/s)
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
All packages are up to date.
W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: tor+http://apow7mjfryruh65chtdydfmqfpj5btws7nbocgtaovhvezgccyjazpqd.onion/torproject.org bullseye InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 74A941BA219EC810
W: Failed to fetch tor+http://apow7mjfryruh65chtdydfmqfpj5btws7nbocgtaovhvezgccyjazpqd.onion/torproject.org/dists/bullseye/InRelease  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 74A941BA219EC810
W: Some index files failed to download. They have been ignored, or old ones used instead.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Calculating upgrade... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
2 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Setting up nginx-core (1.18.0-6.1+deb11u2) ...
Job for nginx.service failed because the control process exited with error code.
See "systemctl status nginx.service" and "journalctl -xe" for details.
invoke-rc.d: initscript nginx, action "start" failed.
● nginx.service - A high performance web server and a reverse proxy server
     Loaded: loaded (/lib/systemd/system/nginx.service; enabled; vendor preset: enabled)
    Drop-In: /etc/systemd/system/nginx.service.d
             └─raspiblitz.conf
     Active: activating (auto-restart) (Result: exit-code) since Tue 2022-07-12 15:59:50 BST; 16ms ago
       Docs: man:nginx(8)
    Process: 1110517 ExecStartPre=/usr/sbin/nginx -t -q -g daemon on; master_process on; (code=exited, status=1/FAILURE)
        CPU: 34ms
dpkg: error processing package nginx-core (--configure):
 installed nginx-core package post-installation script subprocess returned error exit status 1
dpkg: dependency problems prevent configuration of nginx:
 nginx depends on nginx-core (<< 1.18.0-6.1+deb11u2.1~) | nginx-full (<< 1.18.0-6.1+deb11u2.1~) | nginx-light (<< 1.18.0-6.1+deb11u2.1~) | nginx-extras (<< 1.
18.0-6.1+deb11u2.1~); however:
  Package nginx-core is not configured yet.
  Package nginx-full is not installed.
  Package nginx-light is not installed.
  Package nginx-extras is not installed.
 nginx depends on nginx-core (>= 1.18.0-6.1+deb11u2) | nginx-full (>= 1.18.0-6.1+deb11u2) | nginx-light (>= 1.18.0-6.1+deb11u2) | nginx-extras (>= 1.18.0-6.1+
deb11u2); however:
  Package nginx-core is not configured yet.
  Package nginx-full is not installed.
  Package nginx-light is not installed.
  Package nginx-extras is not installed.

dpkg: error processing package nginx (--configure):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 nginx-core
 nginx
E: Sub-process /usr/bin/dpkg returned an error code (1)
```
```
₿ sudo apt remove nginx nginx-core -y
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following packages were automatically installed and are no longer required:
  geoip-database libgeoip1 libnginx-mod-http-geoip libnginx-mod-http-image-filter libnginx-mod-http-xslt-filter libnginx-mod-mail libnginx-mod-stream
  libnginx-mod-stream-geoip nginx-common
Use 'sudo apt autoremove' to remove them.
The following packages will be REMOVED:
  nginx nginx-core
0 upgraded, 0 newly installed, 2 to remove and 0 not upgraded.
2 not fully installed or removed.
After this operation, 1,393 kB disk space will be freed.
(Reading database ... 96050 files and directories currently installed.)
Removing nginx (1.18.0-6.1+deb11u2) ...
Removing nginx-core (1.18.0-6.1+deb11u2) ...
Processing triggers for man-db (2.9.4-2) ...
admin@192.168.2.175:~ ₿ sudo apt install nginx
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  nginx-core
Suggested packages:
  nginx-doc
The following NEW packages will be installed:
  nginx nginx-core
0 upgraded, 2 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/583 kB of archives.
After this operation, 1,393 kB of additional disk space will be used.
Do you want to continue? [Y/n] y
Selecting previously unselected package nginx-core.
(Reading database ... 96040 files and directories currently installed.)
Preparing to unpack .../nginx-core_1.18.0-6.1+deb11u2_arm64.deb ...
Unpacking nginx-core (1.18.0-6.1+deb11u2) ...
Selecting previously unselected package nginx.
Preparing to unpack .../nginx_1.18.0-6.1+deb11u2_all.deb ...
Unpacking nginx (1.18.0-6.1+deb11u2) ...
Setting up nginx-core (1.18.0-6.1+deb11u2) ...
Job for nginx.service failed because the control process exited with error code.
See "systemctl status nginx.service" and "journalctl -xe" for details.
invoke-rc.d: initscript nginx, action "start" failed.
● nginx.service - A high performance web server and a reverse proxy server
     Loaded: loaded (/lib/systemd/system/nginx.service; enabled; vendor preset: enabled)
    Drop-In: /etc/systemd/system/nginx.service.d
             └─raspiblitz.conf
     Active: activating (auto-restart) (Result: exit-code) since Tue 2022-07-12 16:01:34 BST; 38ms ago
       Docs: man:nginx(8)
    Process: 1124408 ExecStartPre=/usr/sbin/nginx -t -q -g daemon on; master_process on; (code=exited, status=1/FAILURE)
        CPU: 39ms

Jul 12 16:01:34 raspberrypi systemd[1]: Failed to start A high performance web server and a reverse proxy server.
dpkg: error processing package nginx-core (--configure):
 installed nginx-core package post-installation script subprocess returned error exit status 1
dpkg: dependency problems prevent configuration of nginx:
 nginx depends on nginx-core (<< 1.18.0-6.1+deb11u2.1~) | nginx-full (<< 1.18.0-6.1+deb11u2.1~) | nginx-light (<< 1.18.0-6.1+deb11u2.1~) | nginx-extras (<< 1.
18.0-6.1+deb11u2.1~); however:
  Package nginx-core is not configured yet.
  Package nginx-full is not installed.
  Package nginx-light is not installed.
  Package nginx-extras is not installed.
 nginx depends on nginx-core (>= 1.18.0-6.1+deb11u2) | nginx-full (>= 1.18.0-6.1+deb11u2) | nginx-light (>= 1.18.0-6.1+deb11u2) | nginx-extras (>= 1.18.0-6.1+
deb11u2); however:
  Package nginx-core is not configured yet.
  Package nginx-full is not installed.
  Package nginx-light is not installed.
  Package nginx-extras is not installed.

dpkg: error processing package nginx (--configure):
 dependency problems - leaving unconfigured
Processing triggers for man-db (2.9.4-2) ...
Errors were encountered while processing:
 nginx-core
 nginx
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

Possible cause:
```
admin@192.168.2.175:~ ₿ sudo nginx -t
nginx: [emerg] a duplicate listen 127.0.0.1:7002 in /etc/nginx/sites-enabled/crtl_tor.conf:4
nginx: configuration file /etc/nginx/nginx.conf test failed

```

**Comment by openoms** [comment]
Edit the files:
`sudo nano /etc/nginx/sites-enabled/crtl_tor.conf`
```
# new setting:
listen 7002;
# old setting:
# listen localhost:7002 ssl http2;
```
and
`sudo nano /etc/nginx/sites-enabled/crtl_tor_ssl.conf`
```
# new:
listen 7003 ssl http2;
# old:
# listen localhost:7003 ssl http2;
```
test with:
`sudo nginx -t`

restart nginx:
```
sudo systemctl restart nginx
```

**Comment by openoms** [comment]
The nginx apt update is succedeing after `sudo nginx -t` does.

Will need to review all nginx config file for deprecated options.

**Comment by openoms** [comment]
Got another report of this: https://t.me/raspiblitz/106863

**Comment by openoms** [comment]
Fixed in [ba657c4](https://github.com/rootzoll/raspiblitz/commit/ba657c49cd5db30dad41ba1e6e8304b360fd345c)
might need to cherry-pick to the v1.7 branch if there are reports of problems after nginx is updated with apt.
On an older node:
```
₿ nginx -v
nginx version: nginx/1.14.2
```
On v1.8.0rc3:
```
₿ nginx -v
nginx version: nginx/1.18.0
```
